### PR TITLE
fix: resolve initialization script paths starting ~

### PR DESCRIPTION
**Pull request by alexmalins**
Fix bug to properly resolve initialization script paths starting with `~` (i.e. user's home dir) supplied to `--init-path` for DuckDB & SQLite adapters.

Using only `.resolve()` means a path like `~/.my_startup.sql` resolves to `/Users/me/~/.my_startup.sql` which does not exist.

Using `.expanduser.resolve()` means the path properly resolves to `/Users/me/.my_startup.sql`.

cf. https://github.com/alexmalins/harlequin-databricks/pull/16

Does this PR require a change to Harlequin's docs?
- [x] No.
- [ ] Yes, and I have opened a PR at [tconbeer/harlequin-web](https://github.com/tconbeer/harlequin-web).
- [ ] Yes; I haven't opened a PR, but the gist of the change is: ...

Did you add or update tests for this change?
- [ ] Yes.
- [x] No, I believe tests aren't necessary.
- [ ] No, I need help with testing this change.

Please complete the following checklist:
- [x] I have added an entry to `CHANGELOG.md`, under the `[Unreleased]` section heading. That entry references the issue closed by this PR.
- [x] I acknowledge Harlequin's MIT license. I do not own my contribution.

### CODE REVIEW

**Comment by alexmalins** [comment]
The reason for the Windows test failures I think is:
- https://knowledge.broadcom.com/external/article/377523/sdwanclient-windows-user-tunnels-are-uns.html
- https://github.com/Alexays/Waybar/issues/3572
- https://github.com/HowardHinnant/date/issues/836#issuecomment-2332581333
- https://lists.iana.org/hyperkitty/list/tz@iana.org/thread/LBVCBNK5NPWBNQ5UDUY3YEZPWA2QV262/
- https://github.com/eggert/tz/commit/812aff32b32c02266361d5538351e0db2a1aee5b

TLDR a bad commit to the IANA TZ data file `northamerica` added a line with `April` instead of `Apr`

The tarball the tests download from [here](https://www.iana.org/time-zones/repository/tzdata-latest.tar.gz) still contains the bad file.

I guess at some point IANA will update that tarball now [commits have been made to fix](https://lists.iana.org/hyperkitty/list/tz@iana.org/message/HSKHZBRALUYDSSSQQ73E5VMJMEWEI4OO/) the April issue.

Until that point would you be happy to ignore the Windows test failures?
Or in another PR we could change the link [here](https://github.com/tconbeer/harlequin/blob/main/src/harlequin/windows_timezone.py#L43) to a version of the tarball without the bad data as a temp fix?

**Comment by tconbeer** [comment]
Thanks, Alex! This is great. Unfortunate about the IANA thing -- thanks for chasing that down. No need to fix that in this moment.
